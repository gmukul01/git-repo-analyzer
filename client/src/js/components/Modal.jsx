import React, { PureComponent } from "react";
import ReactMarkdown from "react-markdown";

export default class Modal extends PureComponent {
  render() {
    const source = this.props.readme || "### Readme is not available";

    return (
      <div className="modal fade" id={this.props.modalId} role="dialog">
        <div className="modal-dialog modal-lg">
          <div className="modal-content">
            <div className="modal-header">
              <button type="button" className="close" data-dismiss="modal">
                &times;
              </button>
              <h4 className="modal-title">{this.props.modalHeader}</h4>
            </div>
            <div className="modal-body">
              <ReactMarkdown source={source} />
            </div>
            <div className="modal-footer">
              <button type="button" className="btn btn-default" data-dismiss="modal">
                Close
              </button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
