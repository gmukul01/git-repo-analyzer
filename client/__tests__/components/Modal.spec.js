import React from "react";
import { shallow } from "enzyme";
import ShallowRenderer from "react-test-renderer/shallow";

import Modal from "components/Modal";

describe("Modal Component", () => {
  let wrapper;
  const initialProps = {
    modalId: "dummyId",
    modalHeader: "dummyHeader"
  };
  it("should match snapshot when source is present in props", () => {
    const props = {
        ...initialProps,
        readme: "### Dummy Readme"
      },
      renderer = new ShallowRenderer(),
      result = renderer.render(<Modal {...props} />);
    expect(result).toMatchSnapshot();
  });

  it("should match snapshot when source is not present in props", () => {
    const renderer = new ShallowRenderer(),
      result = renderer.render(<Modal {...initialProps} />);
    expect(result).toMatchSnapshot();
  });
});
