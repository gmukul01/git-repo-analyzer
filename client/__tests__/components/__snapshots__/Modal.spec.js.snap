// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal page  should match snapshot when source is not present in props 1`] = `
<div
  className="modal fade"
  id="dummyId"
  role="dialog"
>
  <div
    className="modal-dialog modal-lg"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <button
          className="close"
          data-dismiss="modal"
          type="button"
        >
          ×
        </button>
        <h4
          className="modal-title"
        >
          dummyHeader
        </h4>
      </div>
      <div
        className="modal-body"
      >
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="### Readme is not available"
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-default"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Modal page  should match snapshot when source is present in props 1`] = `
<div
  className="modal fade"
  id="dummyId"
  role="dialog"
>
  <div
    className="modal-dialog modal-lg"
  >
    <div
      className="modal-content"
    >
      <div
        className="modal-header"
      >
        <button
          className="close"
          data-dismiss="modal"
          type="button"
        >
          ×
        </button>
        <h4
          className="modal-title"
        >
          dummyHeader
        </h4>
      </div>
      <div
        className="modal-body"
      >
        <ReactMarkdown
          astPlugins={Array []}
          escapeHtml={true}
          plugins={Array []}
          rawSourcePos={false}
          renderers={Object {}}
          skipHtml={false}
          source="### Dummy Readme"
          sourcePos={false}
          transformLinkUri={[Function]}
        />
      </div>
      <div
        className="modal-footer"
      >
        <button
          className="btn btn-default"
          data-dismiss="modal"
          type="button"
        >
          Close
        </button>
      </div>
    </div>
  </div>
</div>
`;
